x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -307,7 +307,7 @@ export function getIndexPatternDatasource({
                 setState={setState}
                 uiSettings={uiSettings}
```

Comment:
Lens aleady has a `KibanaContextProvider` at its root (merged yesterday).
You can use the `useKibana` hook \ `withKibana` HOC instead of passing core components in.

Reply:
@lizozom I found that the `TopNavMenu` component was not actually using the kibana context, it was relying on props to be passed in. This was causing a bug with the PR, and I've fixed it in my latest commit by modifying the `TopNavMenu` to only use kibana context

Reply:
Now it does :)